grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -455,6 +456,15 @@ class StockMovementController {
         render(template: "receipts", model: [receiptItems: receiptItems])
     }
 
+    def events() {
+        StockMovement stockMovement = getStockMovement(params.id)
```

Comment:
I would use the `stockMovementService`'s method to be consistent with the show view page.

Reply:
actually see how the logic is handled for the `def show()` , I would use the same snippet here:
```
// Pull Outbound Stock movement (Requisition based) or Outbound or Inbound Return (Order based)
def stockMovement = outboundStockMovementService.getStockMovement(params.id)
// For inbound stockMovement only
if (!stockMovement) {
    stockMovement =  stockMovementService.getStockMovement(params.id)
}
```

Reply:
I would leave the code as is here, the `getStockMovement` method was introduced not that long ago, and I can see there is an additional logic inside
```
    // Used by SM show page 'tabs' actions - packing list, documents and receipts
    def getStockMovement(String stockMovementId) {
        def stockMovement
        // Pull stock movement in "old fashion" way to bump up performance a bit (instead of getting OutboundStockMovement) for Non-Returns
        def order = Order.get(stockMovementId)
        if (order) {
            stockMovement = outboundStockMovementService.getStockMovement(stockMovementId)
        } else {
            stockMovement = stockMovementService.getStockMovement(stockMovementId)
        }

        return stockMovement
    }
```

Reply:
ok, if this method is used for other tabs as well, I'm fine with that